database-commons/src/main/java/io/cdap/plugin/db/batch/sink/CommonFieldsValidator.java

```diff
@@ -35,35 +36,32 @@
   protected static final Logger LOG = LoggerFactory.getLogger(CommonFieldsValidator.class);
 
   @Override
-  public Set<String> validateFields(Schema inputSchema, ResultSet resultSet) throws SQLException {
+  public void validateFields(Schema inputSchema, ResultSet resultSet, FailureCollector collector) throws SQLException {
     ResultSetMetaData rsMetaData = resultSet.getMetaData();
-
     Preconditions.checkNotNull(inputSchema.getFields());
-    Set<String> invalidFields = new HashSet<>();
     for (Schema.Field field : inputSchema.getFields()) {
       int columnIndex = resultSet.findColumn(field.getName());
       boolean isColumnNullable = (ResultSetMetaData.columnNullable == rsMetaData.isNullable(columnIndex));
       boolean isNotNullAssignable = !isColumnNullable && field.getSchema().isNullable();
+      String name = field.getName();
       if (isNotNullAssignable) {
-        LOG.error("Field '{}' was given as nullable but the database column is not nullable", field.getName());
-        invalidFields.add(field.getName());
+        LOG.error("Field '{}' was given as nullable but the database column is not nullable.", name);
```

Comment:
shouldn't log here. This is similar to how logging and throwing an exception is a bad pattern since it usually results in multiple logs.